watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/posttreatment_ideal_naocl_chlorination_block.py

```diff
@@ -512,20 +512,26 @@ def build_ideal_naocl_mixer_unit(model):
     )
 
     # add new constraint for dosing rate (deactivate constraint for OCl_-)
-    dr = (
-        model.fs.ideal_naocl_mixer_unit.naocl_stream.flow_mol[0].value
-        * model.fs.ideal_naocl_mixer_unit.naocl_stream.mole_frac_comp[0, "OCl_-"].value
+    dr = value(
+        model.fs.ideal_naocl_mixer_unit.naocl_stream.flow_mol[0]
+        * model.fs.ideal_naocl_mixer_unit.naocl_stream.mole_frac_comp[0, "OCl_-"]
+        * model.fs.ideal_naocl_thermo_params.get_component("OCl_-").mw
+    )
+    model.fs.ideal_naocl_mixer_unit.dosing_rate = Var(
+        initialize=dr,
+        domain=NonNegativeReals,
+        units=pyunits.kg / pyunits.s,
     )
-    dr = dr * 74.44 * 1000
-    model.fs.ideal_naocl_mixer_unit.dosing_rate = Var(initialize=dr)
 
     def _dosing_rate_cons(blk):
-        return (
-            blk.dosing_rate
-            == blk.naocl_stream.flow_mol[0]
+        return blk.dosing_rate == (
+            blk.naocl_stream.flow_mol[0]
             * blk.naocl_stream.mole_frac_comp[0, "OCl_-"]
-            * 74.44
-            * 1000
+            * blk.naocl_stream_state[0].params.get_component("OCl_-").mw
+        ) + (
+            blk.naocl_stream.flow_mol[0]
+            * blk.naocl_stream.mole_frac_comp[0, "Na_+"]
+            * blk.naocl_stream_state[0].params.get_component("Na_+").mw
```

Comment:
@adam-a-a this is where the units changed. Given the pretreatment example I think this was always meant to be in `kg/s`, but I'm not certain.

Reply:
Got it.